ts/tree/dbTree.ts

```diff
@@ -1,17 +1,210 @@
-import { Hashable } from "../interfaces";
-import { Hasher } from "./hasher";
-import { Leaf, LeafFactoryFunc } from "./leaves/Leaf";
-import { MemoryTree } from "./memoryTree";
-
-export class DBTree<LeafType extends Leaf<Hashable>> extends MemoryTree {
-    private readonly leafFactory: LeafFactoryFunc<LeafType>;
-
-    constructor(
-        depth: number,
-        leafFactory: LeafFactoryFunc<LeafType>,
-        hasher?: Hasher
-    ) {
-        super(depth, hasher || Hasher.new());
-        this.leafFactory = leafFactory;
+import {
+    BadMergeAlignment,
+    EmptyArray,
+    ExceedTreeSize,
+    MismatchHash,
+    MismatchLength,
+    NegativeIndex
+} from "../exceptions";
+import { Hasher, Node } from "./hasher";
+import { ItemNode } from "./leaves/Node";
+import { AsyncTree, Data, Witness } from "./tree";
+
+export class DBTree implements AsyncTree {
+    public readonly zeros: Array<Node>;
+    public readonly depth: number;
+    public readonly setSize: number;
+    public readonly hasher: Hasher;
+    public readonly nodeType: string;
+
+    public static new(depth: number, nodeType: string, hasher?: Hasher) {
+        return new DBTree(depth, nodeType, hasher || Hasher.new());
+    }
+
+    constructor(depth: number, nodeType: string, hasher: Hasher) {
+        this.depth = depth;
+        this.hasher = hasher;
+        this.zeros = this.hasher.zeros(depth);
+        this.setSize = 2 ** this.depth;
+        this.nodeType = nodeType;
+    }
+
+    public async root(): Promise<Node> {
+        return await this.getNode(0, 0);
```

Comment:
We can add a `_root` variable to cache the root to make the `root` function stay sync. So that we don't have to change all over the codebase to use async function.

The `_root` should be updated in the `ascend` function.

Reply:
set the default value zeros[0] and create a function to set the root during restart